src/Microsoft.Health.Fhir.Shared.Api/Features/ContentTypes/ContentTypeService.cs

```diff
@@ -82,23 +83,30 @@ public async Task CheckRequestedContentTypeAsync(HttpContext httpContext)
                     }
                 }
             }
+
+            string prettyParameterValue = GetParameterValueFromQueryString(httpContext, KnownQueryParameterNames.Pretty);
+
+            if (prettyParameterValue != null && !bool.TryParse(prettyParameterValue, out _))
+            {
+                throw new BadRequestException(Api.Resources.UnsupportedPrettyParameter);
+            }
         }
 
-        private static string GetFormatFromQueryString(HttpContext context)
+        private static string GetParameterValueFromQueryString(HttpContext context, string parameterName)
         {
             EnsureArg.IsNotNull(context, nameof(context));
 
-            // If executing in a rethrown error context, ensure we carry the specified format
+            // If executing in a rethrown error context, ensure we carry the specified formatting.
```

Comment:
nit: should `formatting` be something like `query string value`?

Reply:
Yes that definitely makes more sense. Fixed in dbcf721.